addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -474,6 +477,107 @@ def test_mailing_shortener(self):
+            patch.object(MassMailing, '_get_mass_mailing_immediate_send_nbr', return_value=3):
+            (mailing_1 | mailing_2).action_launch()
+        self.assertEqual(mailing_1.state, 'done')
+        self.assertEqual(mailing_2.state, 'in_queue')
+        self.assertEqual(len(self._mails), 3)
+        # test that mailing_1 mails are sent only
+        for partner in res_partners[:3]:
+            self.assertMailMail(partner, 'sent', author=self.env.user.partner_id, fields_values={'subject': 'One'})
+
+        # enqueue emails if specified limit is set to 0
+        mailing_3 = self.env['mailing.mailing'].create({
+            'name': 'Three',
+            'subject': 'Three',
+            'mailing_model_name': 'mailing.list',
+            'contact_list_ids': [(4, self.mailing_list_1.id), (4, self.mailing_list_2.id)],
+            'body_html': 'This is mass mail marketing demo',
+        })
+        with self.mock_mail_gateway(mail_unlink_sent=False), \
+            patch.object(MassMailing, '_get_mass_mailing_immediate_send_nbr', return_value=0):
+            mailing_3.action_launch()
+        self.assertEqual(mailing_3.state, 'in_queue')
+        self.assertEqual(len(self._new_mails.exists()), 0)
+        self.assertNotSentEmail()
+
+    @users('user_marketing')
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    def test_mailing_action_retry_failed_flow(self):
+        """ Test that mail should be sent directly by action_retry_failed method if total recipients
+        are less than value specified by config param 'mass_mailing_immediate_send_nbr',
+        otherwise it should enqueue the mailing """
+        res_partners = self._create_partner()
+        failed_partner_id = res_partners[0]
+        def write(self, vals):
+            if self.recipient_ids.email == failed_partner_id.email and vals.get('state') != 'exception':
+                return False
+            return models.Model.write(self, vals)
+
+        # number of emails are less then specified limit
+        mailing_1 = self.env['mailing.mailing'].create({
+            'name': 'TestMailingOne',
+            'subject': 'One',
+            'body_html': 'This is mass mail marketing demo',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+            'mailing_domain': [('id', 'in', res_partners[:3].ids)],
+        })
+
+        # number of emails are more then specified limit
+        mailing_2 = self.env['mailing.mailing'].create({
+            'name': 'TestMailingTwo',
+            'subject': 'Two',
+            'body_html': 'This is mass mail marketing demo',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+            'mailing_domain': [('id', 'in', res_partners.ids)],
+        })
+
+        with self.mock_mail_gateway(mail_unlink_sent=False), \
+            patch.object(MassMailing, '_get_mass_mailing_immediate_send_nbr', return_value=3), \
+                patch.object(MailMail, 'write', write):
```

Comment:
still an indentation issue (the patch keyword must be aligned)

check also other occurence
```suggestion
	        with self.mock_mail_gateway(mail_unlink_sent=False), \
             patch.object(MassMailing, '_get_mass_mailing_immediate_send_nbr', return_value=3), \
              patch.object(MailMail, 'write', write):
```

Reply:
![screenshot_spect](https://user-images.githubusercontent.com/54316226/159871321-e2bf82e0-1e0f-458a-912f-78ac2a27278e.png)

it should look like this :) (all keywords aligned)

Reply:
@std-odoo ,
I am getting lint error here(see this image).
![image](https://user-images.githubusercontent.com/72728567/160333757-87a6ea63-91a6-4503-ae6a-a864e370199b.png)

Reply:
yes because keyword are not aligned

Reply:
Here you go
![image](https://user-images.githubusercontent.com/72728567/160341286-55eefef6-a31d-4692-968f-bc18708ef6bf.png)

Reply:
which linter do you use ? flake8 ?

Reply:
I am using pep8 because flake8 is not showing proper error in my ide.

Reply:
I use flake8 for sublime and everything is fine... But ok, align them as you want, not a big deal anyway

Reply:
thank you :)